packages/chat-ui/src/chat/chat-input.tsx

```diff
@@ -131,30 +133,36 @@ function ChatInputForm(props: ChatInputFormProps) {
 function ChatInputField(props: ChatInputFieldProps) {
   const { input, setInput } = useChatUI()
   const { handleKeyDown, setIsComposing } = useChatInput()
-  const type = props.type ?? 'textarea'
-
-  if (type === 'input') {
-    return (
-      <Input
-        name="input"
-        placeholder={props.placeholder ?? 'Type a message'}
-        className={cn(props.className, 'min-h-0')}
-        value={input}
-        onChange={e => setInput(e.target.value)}
-      />
-    )
+  const textareaRef = useRef<HTMLTextAreaElement>(null)
+
+  // auto resize the textarea based on the content
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInput(e.target.value)
+    if (textareaRef.current) {
+      textareaRef.current.style.height = 'auto'
+      let newHeight = Math.max(textareaRef.current.scrollHeight, 100)
+      if (textareaRef.current.scrollHeight > 80) {
+        newHeight += 40 // offset for the textarea padding
+      }
+      textareaRef.current.style.height = `${newHeight}px`
+    }
   }
 
   return (
     <Textarea
+      ref={textareaRef}
       name="input"
-      placeholder={props.placeholder ?? 'Type a message'}
-      className={cn(props.className, 'h-[40px] min-h-0 flex-1')}
+      placeholder={props.placeholder ?? 'Type a message...'}
+      className={cn(
+        props.className,
+        'bg-secondary h-[100px] max-h-[400px] min-h-0 flex-1 resize-none overflow-y-auto rounded-2xl p-4'
+      )}
```

Comment:
```suggestion
      className={cn(
        'bg-secondary h-[100px] max-h-[400px] min-h-0 flex-1 resize-none overflow-y-auto rounded-2xl p-4',
         props.className,
      )}
```
seems to my that overwrites must be passed last, see https://grok.com/share/bGVnYWN5_1d9ea455-0ff8-4acf-8619-ee2f8202ccab